### PR TITLE
Disable volunteer button when patient is discharged

### PR DESCRIPTION
**Pull request by Ashesh3**
Related #3272 
Related Backend Change: https://github.com/coronasafe/care/pull/959

image

Note: This PR can be tested independent of the backend PR, it's just a supplementary.

### CODE REVIEW

File: src/Components/Patient/PatientHome.tsx [hunk 1]
```diff
@@ -1315,16 +1315,49 @@ export const PatientHome = (props: any) => {
...
+                }}
               >
-                <div className="bg-white rounded-lg shadow p-4 h-full space-y-2 hover:bg-gray-200 hover:cursor-pointer">
-                  <div className="text-green-700 text-center">
+                <div
+                  className={`bg-white rounded-lg shadow p-4 h-full space-y-2 ${
+                    !patientData.is_active ||
+                    !(patientData?.last_consultation?.facility === facilityId)
+                      ? " hover:cursor-not-allowed "
+                      : " hover:bg-gray-200 hover:cursor-pointer "
```
Comment (developedBySJ) [inline src/Components/Patient/PatientHome.tsx]: Use `clsx` for conditional classes

**Review by developedBySJ — CHANGES_REQUESTED** [review]

File: src/Components/Patient/PatientHome.tsx (hunk 1 above)
Comment (Ashesh3) [reply src/Components/Patient/PatientHome.tsx]: Done!

**Comment by nihal467** [comment]
image

@Ashesh3 the error persist!

**Comment by Ashesh3** [comment]
@nihal467 This PR just disables the button to add new volunteer for discharged patient, the backend change fixing the issue (removing volunteer when discharged) is a separate PR, not related to this, https://github.com/coronasafe/care/pull/959

**Comment by nihal467** [comment]
test approved

File: src/Components/Patient/PatientHome.tsx [hunk 2]
```diff
@@ -1315,16 +1316,55 @@ export const PatientHome = (props: any) => {
               </div>
               <div
                 className="w-full"
-                onClick={() => setOpenAssignVolunteerDialog(true)}
+                onClick={() => {
+                  if (
+                    !(
+                      !patientData.is_active ||
+                      !(patientData?.last_consultation?.facility === facilityId)
+                    )
```
Comment (khavinshankar) [inline src/Components/Patient/PatientHome.tsx]:
```suggestion
                    !(
                      !patientData.is_active ||
                      !(patientData?.last_consultation?.facility === facilityId)
                    )
```
It seems that this snippet of code is being repeated multiple times, can it be made into a function?

**Review by khavinshankar — CHANGES_REQUESTED** [review]

File: src/Components/Patient/PatientHome.tsx (hunk 2 above)
Comment (Ashesh3) [reply src/Components/Patient/PatientHome.tsx]: Done!

**Review by khavinshankar — APPROVED** [review]

**Review by developedBySJ — APPROVED** [review]
LGTM

**Pull request merged**